### PR TITLE
Allow join sets of different attributtes getting only the common ones…

### PR DESCRIPTION
**Pull request by pconesa**
…. Warn about it.

### CODE REVIEW

**Comment by pconesa** [comment]
Closes #1493

File: pyworkflow/object.py [hunk 1]
```diff
@@ -129,12 +129,30 @@ def getAttributesToStore(self):
...
+                print ("Object.getAttributesToStore: attribute '%s' seems to "
+                      "be overwritten," % key)
+                print ("   since '_objDoStore' was not found. "
+                       "Ignoring attribute. ")
             else:
                 if attr is not None and attr._objDoStore:
                     yield (key, attr)
-            
+
+    def getAttributesToStoreEx(self, prefix=""):
```
Comment (delarosatrevin) [inline pyworkflow/object.py]: @pconesa I think we don't need an extra function in Object. Have you checked Object.getObjDict? the keys of this dictionary are (almost) the strings returned by this new function. I think will be better to reuse the getObjDict function and avoid to add additional ones that are used only in very very specific case.

Comment (pconesa) [reply pyworkflow/object.py]: I'm looking at it...Actually, I didn't put it there initially...only after @josuegbl suggestion I move it there.

Comment (delarosatrevin) [reply pyworkflow/object.py]: My point is that maybe the function is not needed at all.

Comment (pconesa) [reply pyworkflow/object.py]: I think it returns exactly what I need...thanks @delarosatrevin

**Comment by pconesa** [comment]
@delarosatrevin , function removed.

**Comment by delarosatrevin** [comment]
Thanks a lot! I will continue reviewing this PR

**Pull request merged**